accelerator/multi_region_modules/hubNetworking.bicep

```diff
@@ -0,0 +1,1164 @@
+metadata name = 'ALZ Bicep - Hub Networking Module'
+metadata description = 'ALZ Bicep Module used to set up Hub Networking'
+
+type subnetOptionsType = ({
+  @description('Name of subnet.')
+  name: string
+
+  @description('IP-address range for subnet.')
+  ipAddressRange: string
+
+  @description('Id of Network Security Group to associate with subnet.')
+  networkSecurityGroupId: string?
+
+  @description('Id of Route Table to associate with subnet.')
+  routeTableId: string?
+
+  @description('Name of the delegation to create for the subnet.')
+  delegation: string?
+})[]
+
+type lockType = {
+  @description('Optional. Specify the name of lock.')
+  name: string?
+
+  @description('Optional. The lock settings of the service.')
+  kind: ('CanNotDelete' | 'ReadOnly' | 'None')
+
+  @description('Optional. Notes about this lock.')
+  notes: string?
+}
+
+@sys.description('The Azure Region to deploy the resources into.')
+param parLocation string = resourceGroup().location
+
+param parSecondaryLocation string?
```

Comment:
I think if we will have an additional parameters file for the secondary region, we won't need to a have an additional `parSeconadaryLocation` one since this module will deploy just one additional region. If we decide to use this to deploy N-regions then we can have this parameter as an array and loop through it as you already have here